healsparse/healSparseMap.py

```diff
@@ -358,4 +357,84 @@ def generateHealpixMap(self, nside=None, reduction='mean'):
         """
 
         pass
+ 
+    def degrade(self, nside_out, reduction='mean'):
+        """
+        Reduce the resolution, i.e., increase the pixel size
+        of a given sparse map
+        
+        Args:
+        ----
+        nside_out: `int`, output Nside resolution parameter.
+        reduction: `str`, reduction method (mean, median, std, max, min).
+        """ 
+        
+        if self._nsideSparse < nside_out:
+            raise ValueError('nside_out should be smaller than nside for the sparseMap')
+        # Count the number of filled pixels in the coverage mask
+        npop_pix = np.count_nonzero(self.coverageMask)
+        # We need the new bitShifts and we have to build a new CovIndexMap
+        bitShift = 2 * int(np.round(np.log(nside_out / self._nsideCoverage) / np.log(2)))
+        nFinePerCov = 2**bitShift 
+        # Work with RecArray (we have to change the resolution to all maps...)
+        if self._isRecArray:
+            dtype = []
+            # We should avoid integers
+            for key, value in self._sparseMap.dtype.fields.items():
+                if isinstance(self._sparseMap[key],int):
+                    dtype.append((key,np.float))
+                else:
+                    dtype.append((key,value[0]))
+            dtype = self._sparseMap.dtype
+            # Allocate new map
+            newsparseMap = np.zeros((npop_pix+1)*nFinePerCov, dtype=dtype)
+            for key, value in newsparseMap.dtype.fields.items():  
+                aux = self._sparseMap[key]
+                aux = aux.reshape((npop_pix+1, (nside_out//self._nsideCoverage)**2, -1))
+                wgt = np.ones_like(aux, dtype=np.float)
```

Comment:
I think this should work (and it does in the tests) but I think we can use the same code as below (`nanmean` etc), we just need to make sure that `aux = self._sparseMap[key].astype(np.float)` and then filling with `NaN`s.  Then we can get the median functionality as well.